modules/sc-mesh-secure-deployment/src/2_0/features/cbma/secure_channel/secchannel.py

```diff
@@ -0,0 +1,74 @@
+import queue
+import ssl
+import socket
+import json
+import threading
+from tools.custom_logger import CustomLogger
+
+logger_instance = CustomLogger("SecChannel")
+
+
+class SecMessageHandler:
+    def __init__(self, socket, shutdown_event=threading.Event()):
+        self.socket = socket
+        self.logger = logger_instance.get_logger()
+        self.callback = None
+        self.shutdown_event = shutdown_event
+
+    def set_callback(self, callback):
+        """Set the callback function to be executed when a message is received."""
+        self.callback = callback
+
+    def _is_socket_active(self):
+        """Check if the socket is active."""
+        try:
+            # Just a simple check; will rely on recv() in receive_message() for more detailed check.
+            return self.socket.fileno() != -1
+        except socket.error:
+            return False
+
+    def _is_ssl_socket(self):
+        return isinstance(self.socket, ssl.SSLSocket)
+
+    def send_message(self, message):
+        """Send a message through the socket."""
+        if not self._is_ssl_socket():
+            self.logger.error("Socket is not SSL enabled.")
+            return
+
+        if not self._is_socket_active():
+            self.logger.error("Socket is not active.")
+            return
+
+        try:
+            self.socket.send(message.encode())
```

Comment:
send() - don't guarantee complete packet send.

https://docs.python.org/3/howto/sockets.html#using-a-socket

You need to have known message sizes. Use sendall() to send complete packet and on receiver side you need to know how long packet is coming.